pkg/volume/provision.go

```diff
@@ -370,34 +376,56 @@ func (p *nfsProvisioner) getServer() (string, error) {
 		return "", fmt.Errorf("hostname -i had bad output %s, no address to use", string(out))
 	}
 
-	nodeName := os.Getenv(p.nodeEnv)
-	if nodeName != "" {
-		glog.Infof("using node name %s=%s as NFS server IP", p.nodeEnv, nodeName)
-		return nodeName, nil
-	}
+	// we want the persistentvolume to connect to the "servicename.namespace" hostname, so that
+	// it will survive any restarts of the service, pod or node on which the nfs server happens 
+	// to be running (in previous versions we returned the service IP here, which was a bit less 
+	// stable in case of changes or to the service)
+	namespace := os.Getenv(p.namespaceEnv)
+	serviceName := os.Getenv(p.serviceEnv)
 
+	// we already fetch the podIp because we need it a number of times
 	podIP := os.Getenv(p.podIPEnv)
-	if podIP == "" {
-		return "", fmt.Errorf("pod IP env %s must be set even if intent is to use service cluster IP as NFS server IP", p.podIPEnv)
-	}
 
-	serviceName := os.Getenv(p.serviceEnv)
-	if serviceName == "" {
-		glog.Infof("using potentially unstable pod IP %s=%s as NFS server IP (because neither service env %s nor node env %s are set)", p.podIPEnv, podIP, p.serviceEnv, p.nodeEnv)
-		return podIP, nil
+	// In the unlikely case that we do not have a namespace or servicename, we will try some alternatives
+	// so that we have _at least_ a volume that we can offer to the caller (although not a very stable one)
+	// TODO Is it clever to use these alternatives in the first place instead of just crashing? Handing
+	//      out such an unstable server name is a bit like waiting for a future disaster that will hit somebody
+	//      when nobody is around to solve it.   
```

Comment:
I'd prefer to crash with informative message, instead of silently using unstable IPs. But is it valid only in case if we running nfs-server-provisioner outside of the pod? Now I'm bit confused.